app/routes.py

```diff
@@ -1 +1,172 @@
-from flask import Blueprint
\ No newline at end of file
+#/task-list-api/app/routes.py
+from flask import Blueprint, request, jsonify
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from datetime import datetime
+from slack import send_slack_notification
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route('/<int:task_id>', methods=['GET'])
+def get_task(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        return jsonify(task=task.to_dict()), 200
+    else:
+        return jsonify(details="Task not found"), 404
```

Comment:
There are quite a few places where we need to query for a task and check to make sure that it exists. It would be great to have all of this in a helper function like we did in class.